### DISCUSSION TITLE
[Help] How to plot FUNGuild "trans_func" results by Group instead of modules

### DISCUSSION THREAD

**Discussion by ignerocha-f** [ChiLiubio/microeco#378] [Q&A]
First of all, thank you for this very useful package for microbiome analysis!

I am taking advantage of the "trans_func" functions for ITS data and I would like to test FUNGuild and FungalTraits on my data. However, I would like to express the results by group/treatment that is in my metadata instead of modules, as described in the example "Other dataset > 9.1 Fungi data".

I know this may be a silly question, but I am unable to convert the "t2$cal_spe_func_perc" object into something that I can use to generate a relative abundance plot of functions by "Group".

Could someone help me?
Thank you very much in advance!

**Comment by ignerocha-f** [comment]
Sorry, I guess the answer is on the "14.9 FAPROTAX + differential test" in the tutorial. 

Thank you so much!

**Comment by ChiLiubio** [comment]
Yes. To make it easy to find the test, I will add a link in the Fungi part. Thanks.

**Reply by ignerocha-f** [reply]
Thanks for that, @ChiLiubio!

One more question, please. 
Is that possible to filter the functional groups by "confidenceRanking"?

I tryed the following commands, but although it filtered the "Highly Probable" ones in the "res_spe_func_raw_funguild", it doesn't select the groups in the "res_spe_func_perc" table. 

t2$cal_spe_func_perc(abundance_weighted = T)
t2$res_spe_func_raw_funguild <- subset(t2$res_spe_func_raw_funguild, confidenceRanking == "Highly Probable")
tmp_mt <- clone(meco_fungi)
tmp_mt$taxa_abund$func <- as.data.frame(t(t2$res_spe_func_perc), check.names = FALSE)

Doing this, groups with less confidenceRanking is still in the final plot. 

I'd appreciate your help on it! 
Thank you!

**Reply by ChiLiubio** [reply]
Hi. I add a new parameter FUNGuild_confidence  in the cal_spe_func function to select required "confidenceRanking". Please reinstall the microeco package from github and try again. The CRAN version 1.8.0 will be released in the coming days.
```
library(microeco)
library(magrittr)
# load ITS data
data(sample_info_ITS)
data(otu_table_ITS)
data(taxonomy_table_ITS)
# create microtable object
meco_fungi <- microtable$new(sample_table = sample_info_ITS, otu_table = otu_table_ITS, tax_table = taxonomy_table_ITS)
# remove the taxa not assigned in the Kingdom "k__Fungi"
meco_fungi$tax_table %<>% base::subset(Kingdom == "k__Fungi")
# use tidy_dataset() to make OTUs and samples information consistent across files
meco_fungi$tidy_dataset()

t1 <- trans_func$new(meco_fungi)
# default  FUNGuild_confidence = c("Highly Probable", "Probable", "Possible")
t1$cal_spe_func(fungi_database = "FUNGuild")

t2 <- trans_func$new(meco_fungi)
# select "Highly Probable"
t2$cal_spe_func(fungi_database = "FUNGuild", FUNGuild_confidence = c("Highly Probable"))
t2$cal_spe_func_perc(abundance_weighted = FALSE)
```

**Reply by ignerocha-f** [reply]
I can't thank you enough for your prompt response to my question. Now it's wonderful! 
Thank you very much for that!

In terms of the statistical test, is up to me to choose to use rf and lefse or is it recommended to stick with anova?

I really appreciate your effort on keep improving this project!

**Reply by ChiLiubio** [reply]
Thanks! Yes. It is up to you.
